`TestValidatorSetHandling` is flaky

`TestValidatorSetHandling` is flaky - it sometimes fails like this:
```
    client_test.go:709: 
        	Error Trace:	client_test.go:709
        	Error:      	Received unexpected error:
        	            	failed to load validators for height 13: key not found
        	Test:       	TestValidatorSetHandling
I[2022-07-01|21:21:07.297] indexed block                                module=txindex height=13
    client_test.go:710: 
        	Error Trace:	client_test.go:710
        	Error:      	Expected value not to be nil.
        	Test:       	TestValidatorSetHandling
--- FAIL: TestValidatorSetHandling (0.13s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x11fa485]

goroutine 1972 [running]:
testing.tRunner.func1.2({0x134e020, 0x23f2590})
	/usr/lib/go/src/testing/testing.go:1389 +0x24e
testing.tRunner.func1()
	/usr/lib/go/src/testing/testing.go:1392 +0x39f
panic({0x134e020, 0x23f2590})
	/usr/lib/go/src/runtime/panic.go:838 +0x207
github.com/celestiaorg/optimint/rpc/client.TestValidatorSetHandling(0xc05f6a7380?)
	/home/tomek/devel/celestia/optimint/rpc/client/client_test.go:711 +0x1625
testing.tRunner(0xc05f6a7380, 0x183fa70)
	/usr/lib/go/src/testing/testing.go:1439 +0x102
created by testing.(*T).Run
	/usr/lib/go/src/testing/testing.go:1486 +0x35f
```

--- a/block/manager.go
+++ b/block/manager.go
@@ -401,9 +401,6 @@ func (m *Manager) publishBlock(ctx context.Context) error {
 		return err
 	}
 
-	// Only update the stored height after successfully submitting to DA layer
-	m.store.SetHeight(block.Header.Height)
-
 	// Commit the new state and block which writes to disk on the proxy app
 	_, _, err = m.executor.Commit(ctx, newState, block, responses)
 	if err != nil {
@@ -431,6 +428,9 @@ func (m *Manager) publishBlock(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
+
+	// Only update the stored height after successfully submitting to DA layer and committing to the DB
+	m.store.SetHeight(block.Header.Height)
 
 	m.publishHeader(block)
 

